app/controllers/tweets_controller.rb

```diff
@@ -42,22 +42,30 @@ def create
   end
 
   def edit
-    @tweet = Tweet.find_by(id: params[:id])
+    if current_user.admin
+      @tweet = Tweet.find_by(id: params[:id])
+    else
+      redirect_to lessons_url
```

Comment:
Redirecting if a user is not an admin seems to be best handled by a `before_action` filter. See [the RubyGuides Aciton Controller Overview](http://guides.rubyonrails.org/action_controller_overview.html#filters).